src/locktower.rs

```diff
@@ -0,0 +1,172 @@
+use std::collections::VecDeque;
+pub struct VoteLock {
+    /// Fork id, same as slot id. There can only be 1 fork per slot, since the empty PoH ticks
+    /// cannot be voted on.  A validator cannot vote on a fork unless that fork is connected to
+    /// a trunk. The starting trunk is the initial `finalized_fork`.
+    fork: u64,
```

Comment:
Why not the type `Hash`?

Reply:
@garious  how do we do addition and subtraction on Hash?

Reply:
I suspect there needs to be a separate `block_height` field.